prefect_azure/blob_storage.py

```diff
@@ -159,6 +159,5 @@ def example_blob_storage_list_flow():
     async with blob_storage_credentials.get_container_client(
         container
     ) as container_client:
-        blobs = [blob async for blob in container_client.list_blobs()]
-
+        blobs = [dict(blob) async for blob in container_client.list_blobs()]
```

Comment:
Do we need to update the type hint for this function now that we're returning a list of dicts? Changing the return type will be a breaking change.